WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -66,6 +66,17 @@ final class OrderDetailsViewController: UIViewController {
     ///
     private var cardReaderAvailableSubscription: Combine.Cancellable? = nil
 
+    /// Controller that facilitates connecting to card readers
+    private lazy var cardReaderConnectionController: CardReaderConnectionController = {
```

Comment:
It seems to me like there is potential for a leak here. CardReaderConnectionController holds a reference to OrderDetailsViewController and the other way around.

Is this declared as a property because it needs to be a property, or to encapsulate the initial configuration of the view controller that is being instantiated? Because if we don't need to retain the new view controller (CardReaderConnectionController) as a property, and what we want is to tidy up the way it is initialled, it could be initialized in a private method.

Reply:
I agree. I changed the approach in https://github.com/woocommerce/woocommerce-ios/pull/4721/commits/af4232cacb613e4b77f3cf66ea0d88a4f88aa101 such that the reference to the "from" view controller is discarded at the end of the flow - i.e. after a reader is connected, the user cancels, or an error occurs.